### PR TITLE
feat: add NexusLayout

### PR DESCRIPTION
**Pull request by stephenjwatkins**
## 📝 Changes

- **Different from Spec:** updates `selectedKey` to `selectedHref` to make the API cleaner
- Adds NexusLayout
- Adds support for global client-side routing with links
- _Does not include mobile state yet_
- Some typography will be updated with outstanding typography PR

[Link to story](https://nexus-layout--63f50c7c86f6514d2e0ef4be.chromatic.com/?path=/docs/components-productlayout-nexuslayout--docs).

## ✅ Checklist

Easy UI has certain UX standards that must be met. In general, non-trivial changes should meet the following criteria:

- [x] Visuals match Design Specs in Figma
- [x] Stories accompany any component changes
- [x] Code is in accordance with our style guide
- [x] Design tokens are utilized
- [x] Unit tests accompany any component changes
- [x] TSDoc is written for any API surface area
- [x] Specs are up-to-date
- [x] Console is free from warnings
- [x] No accessibility violations are reported
- [x] Cross-browser check is performed (Chrome, Safari, Firefox)
- [x] Changeset is added

~Strikethrough~ any items that are not applicable to this pull request.

### CODE REVIEW

File: easy-ui-react/src/NexusLayout/NexusLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    styles[variationName("backgroundDecoration", backgroundDecoration)],
+  );
+  return <div className={className}>{children}</div>;
+}
+
+function NexusLayoutHeader(props: NexusLayoutHeaderProps) {
+  const { logo = <EasyPostLogo />, children } = props;
+  return (
+    <header className={styles.header}>
+      <HorizontalStack blockAlign="center" align="space-between">
```
Comment (kevinalexliu) [inline easy-ui-react/src/NexusLayout/NexusLayout.tsx:111]: Question: `children` have both `nav` and `actions` in it, and with `space-between` on `HorizontalStack` will the `nav` be centered on the page if `logo` and `actions` have different widths?

**Comment by OskiTheCoder** [comment]
**question**: out of curiosity, is the intended use case that these will all be menus? the figma suggests that might be the case, but curious if you know for certain. if not, is it possible to compose the components in a way where they serve as nav links?

Screenshot 2024-10-18 at 6 48 27 PM

**Comment by stephenjwatkins** [comment]
> **question**: out of curiosity, is the intended use case that these will all be menus? the figma suggests that might be the case, but curious if you know for certain. if not, is it possible to compose the components in a way where they serve as nav links?

good thought. i created `<NexusLayout.LinkAction />` for this use case in case we run into it

File: easy-ui-react/src/NexusLayout/NexusLayout.tsx (hunk 1 above)
Comment (stephenjwatkins) [reply easy-ui-react/src/NexusLayout/NexusLayout.tsx]: good call. addressed

File: easy-ui-react/src/utilities/EasyPostLogo.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
```
Comment (OskiTheCoder) [inline easy-ui-react/src/utilities/EasyPostLogo.tsx:1]: good call to add the logo as an export in utils

**Review by OskiTheCoder — APPROVED** [review]
LGTM! nice

**Review by kevinalexliu — APPROVED** [review]
LGTM

**Pull request merged**